server/services/SsotToRobotParsing/generateCodeForRpaTasks.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @category Server
+ * @module
+ */
+const {
+  ACTIVITY_IDENTIFIER,
+  FOURSPACE,
+  LINEBREAK,
+  COMMENT,
+} = require('./robotCodeConstants');
+
+/**
+ * @description Checks whether the given element is of type instruction and contains rpa attributes
+ * @param {Object} currentElement Element to check
+ * @returns {Boolean} Value specifies if object is of type instruction and contains rpa attributes
+ */
+const isAnRpaInstruction = (currentElement) =>
+  currentElement.type === ACTIVITY_IDENTIFIER;
+
+/**
+ * @description Will create a prefix to catch the output variable of an activity, if one is present
+ * @param {Object} paramObject The parameter object to check and loop through
+ * @returns {String} String specifying the output variables name
+ */
+const setOutputVar = (paramObject) => {
+  let newCodeLine = FOURSPACE;
+
+  if (paramObject.outputVariable) {
+    newCodeLine += `\${${paramObject.outputVariable}} =${FOURSPACE}`;
+  }
+
+  return newCodeLine;
+};
+
+/**
+ * @description Will append all provided parameters to a string which can be used to generate the RPAf file
+ * @param {Object} parameterObject The parameter object to check and loop through
+ * @returns {String} String specifying the input parameters with the needed spacing in between
+ */
+const appendRpaInputParameter = (parameterObject) => {
+  let newCodeLine = '';
+
+  const sortedInputs = parameterObject.rpaParameters.sort(
+    (a, b) => a.index - b.index
+  );
+  sortedInputs.forEach((parameter) => {
+    // regex will return -1 if no $$text$$ was found
+    newCodeLine +=
+      parameter.value.search(/\$\$(.*?)\$\$/) < 0
+        ? `${FOURSPACE}${parameter.value}`
+        : `${FOURSPACE}$\{${parameter.value.split('$$')[1]}}`;
+  });
+
+  return newCodeLine;
+};
+
+/**
+ * @description Checks whether the given element has a successor element
+ * @param {Object} currentElement Element to check
+ * @returns {Boolean} Value specifies if element has a successor element
+ */
+const successorTasksExist = (currentElement) =>
+  currentElement.successorIds !== undefined &&
+  currentElement.successorIds[0] !== '';
+
+/**
+ * @description Receives an array of all elements and generates the .robot code for the elements recursively.
```

Comment:
Please check the documentation, this doesn't seem to match

Reply:
I guess it's just parameters and attributes missing

Reply:
✅